analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ProgramStateTest.SymbolValue.cs

```diff
@@ -150,8 +151,7 @@ public void SetSymbolConstraint_ExistingValue_PreservesOtherConstraints()
             var sut = ProgramState.Empty
                 .SetSymbolValue(symbol, new SymbolicValue().WithConstraint(TestConstraint.First))
                 .SetSymbolConstraint(symbol, DummyConstraint.Dummy);
-            sut[symbol].HasConstraint(TestConstraint.First).Should().BeTrue("original constraints of different type should be preserved");
-            sut[symbol].HasConstraint(DummyConstraint.Dummy).Should().BeTrue();
+            sut[symbol].Should().HaveOnlyConstraints(new SymbolicConstraint[] { TestConstraint.First, DummyConstraint.Dummy }, "original constraints of different type should be preserved");
```

Comment:
Let's change the signatures to benefit from `ConstraintKind`

Reply:
Let's do it in a follow-up PR. There is also a need for custom assertions for `ProgramState` itself. Let's see what is more urgent.